Add documentation about Apache Solr support (#108)
LangStream/documentation: configuration-resources/data-storage/solr.md

                "type":"knn_vector",
                "stored":true,
                "indexed":true
                }
            }
        - "api": "/schema"
          "body": |
            {
               "add-field":{
                   "name":"text",
                   "type":"string",
                   "stored":true,
                   "indexed":false,
                   "multiValued": false
               }
            }
```

As you can see in the "create-statements" section above, you can configure a number of "commands" that translate to Solr API calls.
You can invoke any of the APIs, for each command you have to declare:
- the "api" you want to call: "/schema" or "/api/collections"
- the body of the HTTP Request, as a JSON string
- you can also set the HTTP method, if you want to use something different than "POST"

If you set "api" to "/schema" then you are using the [Solr Schema API](https://solr.apache.org/guide/solr/latest/indexing-guide/schema-api.html).
If you set "api" to "/api/collections" then you are using the [Solr Collections API](https://solr.apache.org/guide/solr/latest/configuration-guide/collections-api.html).

#### Querying Solr

> When you use the "query-vector-db" agent to query Solr, you can use the following parameters:

Reviewer: ```suggestion
Use the "query-vector-db" agent to query Solr with the following parameters:
```

```yaml
  - name: "lookup-related-documents"
    type: "query-vector-db"
    configuration:
      datasource: "SolrDataSource"
      query: |
        {
          "q": "{!knn f=embeddings topK=10}?"
        }
      fields:
        - "fn:toListOfFloat(value.question_embeddings)"
      output-field: "value.related_documents"
```

As usual you can use the '?' symbol as a placeholder for the fields that you specify in the "q" section.

> For Apache Solr you write in the "query" field a JSON that describes the parameters to pass in the Solr query.

Reviewer: ```suggestion
For Apache Solr, the "query" field requires a JSON that describes the parameters to pass in the Solr query.
```

> Usually you provide a value for "q" that is the main query string, and then you can add other parameters.

Reviewer: ```suggestion
Usually you provide a value for "q" for the main query string, and then add other parameters.
```

> In the example above, we are using the "knn" query parser to perform a [Vector Similarity Search]((https://solr.apache.org/guide/solr/latest/query-guide/dense-vector-search.html)).

Reviewer: ```suggestion
In the example above, we are using the "knn" query parser to perform a [Vector Similarity Search](https://solr.apache.org/guide/solr/latest/query-guide/dense-vector-search.html).
```

### Writing to Solr

> When you use the "vector-db-sink" agent to write to Solr, you can use the following parameters:

Reviewer: ```suggestion
Use the "vector-db-sink" agent to write to Solr with the following parameters:
```

```yaml
  - name: "Write to Solr"
    type: "vector-db-sink"
    input: chunks-topic
    configuration:
      datasource: "SolrDataSource"
      collection-name: "documents"
      fields:
        - name: "id"
          expression: "fn:concat(value.filename, value.chunk_id)"
        - name: "embeddings"
          expression: "fn:toListOfFloat(value.embeddings_vector)"
        - name: "text"
          expression: "value.text"
````

> Set the collection-name to the name of the collection you want to write to. Then you define the fields in the "fields" list. This works similarly to the ['compute' agent](../../pipeline-agents/data-transform/compute.md), you can define the name of the field and the expression to compute the value of the field.

Reviewer: ```suggestion
Set the collection-name to the name of the collection you want to write to, and define the fields in the "fields" list. This works similarly to the ['compute' agent](../../pipeline-agents/data-transform/compute.md), where you define the name of the field and the expression to compute the value of the field.
```
